File: FasdtApi/llm_config.py
from pydantic  import Field,BaseModel

from fastapi import APIRouter, Depends
from service.exceptions import InvalidInput, NotFound
from typing import Optional
from models.init_db import User
from models.async_db import get_async_db
from service.dependencies import get_current_user_async
from service.llm import llm_config_service
from service.llm.model_catalog import supported_models as supported_model_catalog

router =APIRouter(prefix="/llm_config", tags=["模型配置"])

class ConfigSave(BaseModel):
    model_name: str = Field(min_length=1,max_length=100)
    api_key: str = Field(min_length=1, max_length=500)
    api_url: Optional[str] = Field(default=None, max_length=500)


class QuickConnect(BaseModel):
    provider: str = Field(min_length=1, max_length=40)
    api_key: str = Field(min_length=1, max_length=500)
    capabilities: Optional[list[str]] = Field(default=None)  # ["chat","embedding"]，缺省=全部

@router.get("/list",summary="获取用户的模型配置列表")
async def list_configs(
        async_db=Depends(get_async_db),
        current_user : User =Depends(get_current_user_async)
):
    return await llm_config_service.async_list_configs(async_db, current_user)


@router.get("/supported_models", summary="获取后端支持的聊天模型")
async def supported_models(current_user: User = Depends(get_current_user_async)):
    catalog = supported_model_catalog()
    return {
        "models": [item["model_name"] for item in catalog["chat"] + catalog["embedding"]],
        "chat": catalog["chat"],
        "embedding": catalog["embedding"],
    }
@router.post("",summary="新增或更新模型配置")
async def save_config(
        config :ConfigSave,
        async_db=Depends(get_async_db),
        current_user : User = Depends(get_current_user_async)
):
    return await llm_config_service.async_save_config(
        async_db, current_user, config.model_name, config.api_key, config.api_url
    )


@router.post("/quick_connect", summary="一次连接，多项能力（普通模式）")
async def quick_connect(
        payload: QuickConnect,
        async_db=Depends(get_async_db),
        current_user: User = Depends(get_current_user_async),
):
    """选平台 + 粘一次 Key → 自动配好聊天 + 资料读取两条配置并逐条测试。"""
    return await llm_config_service.async_quick_connect(
        async_db, current_user, payload.provider, payload.api_key, payload.capabilities,
    )


@router.post("/{model_name}/test", summary="测试模型配置")
async def test_config(
        model_name: str,
        async_db=Depends(get_async_db),
        current_user: User = Depends(get_current_user_async)
):
    result = await llm_config_service.async_test_config(async_db, current_user, model_name)
    if not result.get("ok"):
        raise InvalidInput(result)
    return result


@router.delete("/{model_name}",summary="删除模型配置")
async def delete_config(
        model_name:str,
        async_db=Depends(get_async_db),
        current_user = Depends(get_current_user_async)
):
    result = await llm_config_service.async_delete_config_by_model(async_db, current_user, model_name)
    if result["message"] =="配置不存在":
        raise NotFound("配置不存在")
    return result

File: frontend/src/api/llmConfig.ts
import request from '../utils/request'

export interface LlmConfig {
  id: number
  model_name: string
  api_key: string
  api_url?: string | null
  provider?: string
  kind?: 'chat' | 'embedding' | string
  is_active: number
}

export interface LlmConfigPayload {
  model_name: string
  api_key: string
}

export interface SupportedModel {
  model_name: string
  provider: string
  kind: 'chat' | 'embedding'
}

export interface LlmConfigTestResult {
  ok: boolean
  model_name: string
  kind?: 'chat' | 'embedding' | string
  message: string
  elapsed_ms?: number
  dimension?: number
  preview?: string
  error?: string
}

export async function listConfigs(): Promise<LlmConfig[]> {
  const { data } = await request.get('/llm_config/list')
  return data as LlmConfig[]
}

export async function saveConfig(payload: LlmConfigPayload) {
  const { data } = await request.post('/llm_config', payload)
  return data
}

export async function deleteConfig(modelName: string) {
  const { data } = await request.delete(`/llm_config/${encodeURIComponent(modelName)}`)
  return data
}

export async function testConfig(modelName: string): Promise<LlmConfigTestResult> {
  const { data } = await request.post(`/llm_config/${encodeURIComponent(modelName)}/test`)
  return data as LlmConfigTestResult
}

export interface QuickConnectResult {
  provider: string
  saved: string[]
  skipped: Array<{ capability: string; reason: string }>
  results: Record<string, LlmConfigTestResult>
}

/** 一次连接，多项能力：选平台 + 粘一次 Key，后端原子保存聊天+资料两条配置并逐条测试。 */
export async function quickConnect(payload: {
  provider: string
  api_key: string
  capabilities?: Array<'chat' | 'embedding'>
}): Promise<QuickConnectResult> {
  const { data } = await request.post('/llm_config/quick_connect', payload)
  return data as QuickConnectResult
}

export async function listSupportedModelCatalog(): Promise<{
  chat: SupportedModel[]
  embedding: SupportedModel[]
}> {
  const { data } = await request.get('/llm_config/supported_models')
  return {
    chat: data.chat || [],
    embedding: data.embedding || [],
  }
}
